src/ddscxx/include/org/eclipse/cyclonedds/sub/qos/DataReaderQosDelegate.hpp

```diff
@@ -244,6 +246,18 @@ DataReaderQosDelegate::policy<dds::core::policy::ReaderDataLifecycle>()
     return lifecycle_;
 }
 
+
```

Comment:
Please also encapsulate this whole body with the conditional OMG_DDS_EXTENSIBLE_AND_DYNAMIC_TOPIC_TYPE_SUPPORT macro.
(BTW: I see that in this particular case the embedding OMG_DDS_OWNERSHIP_SUPPORT macro was also missing for the "ownership" policy).